sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -745,6 +753,11 @@ static int sof_audio_probe(struct platform_device *pdev)
 	if (sof_rt5682_quirk & SOF_SPEAKER_AMP_PRESENT)
 		sof_audio_card_rt5682.num_links++;
 
+	if (sof_rt5682_quirk & SOF_MAX98373_SPEAKER_AMP_PRESENT) {
+		sof_max98373_codec_conf(&sof_audio_card_rt5682);
+		sof_audio_card_rt5682.name = "tgl_max73rt82";
```

Comment:
why the tgl prefix?
and why change the card name anyways?

Reply:
>why the tgl prefix?
I will remove tgl prefix, add sof_. I will update patches.

> why change the card name anyways?
We are using one OS to support multiple Audio daughter boards (i.e. multiple audio configurations like: TGL with max_98373_rt5682, with max_98357A_rt5682 etc tec). CRAS config folder name is same as sound card name. CRAS config folder will have tuning data and ucm files. They should be unique to have one OS support. So, want to change card name to support relevant audio daughter card/configuration.

Reply:
@sathya-nujella the direction we had from Jaroslav was to use strings that can be retrieved by userspace with 'control components', e.g. spk:max98373 hs:rt5682 so that the card name remains the same instead of being an encoding of the platform configuration

see e.g. commits
0d5c8187562848b619a35f2ffc5e18ce703e9f3d b5706f8ec29fb461571d25e3e813ede121fe31cd 
b5706f8ec29fb461571d25e3e813ede121fe31cd

I am planning on using this for SoundWire, maybe we should do the same here.

@cujomalainey @dgreid FYI on UCM impact

Reply:
We will work with Google to get components_string to be used in CRAS side.. may be I will raise a public chromium and work with Dylan/Curtis.
In new patchset, I updated card name as needed now.
As soon as user space support comes, I can add PR to add components_string and remove this overriding. Please suggest if this is ok to continue until full working frame work is available.

Reply:
@sathya-nujella I suggest you put this name change in a different patch, and we keep the name change in the SOF tree until the rest is complete, while we upstream the rest of your changes. That way we only upstream stuff that works with the component part. Would this work for you?

Reply:
sure @plbossart . Thank you for suggesting alternative ideas.
I have updated patchset now and move to a separate patch only this overriding the card name.

Reply:
@sathya-nujella I just started doing this for the SoundWire card, and that would give something like:

````
root@plb:~# more /sys/class/sound/card0/components 
HDA:8086280b,80860101,00100000 cfg-spk:2 hs:rt711 spk:rt1308 mic:rt715
````
userspace can query/parse that string w/ alsa-lib and know what components are handled by the card. It's a lot more flexible/scalable than using a card name :-)

Reply:
> @sathya-nujella I just started doing this for the SoundWire card, and that would give something like:
> 
> ```
> root@plb:~# more /sys/class/sound/card0/components 
> HDA:8086280b,80860101,00100000 cfg-spk:2 hs:rt711 spk:rt1308 mic:rt715
> ```
> 
> userspace can query/parse that string w/ alsa-lib and know what components are handled by the card. It's a lot more flexible/scalable than using a card name :-)

Sure Pierre. Understood..Thank you!

Reply:
Ack, so this means that all cards using the same codecs will bear the same name? Will we still be able to use the card name or will have to change to the spk: hs: notation for the ucm?

Reply:
sound card name will be same "sof-rt5682" for all machines which ever uses this machine driver "sof-rt5682.c". For CRAS to have unique ucm/cras-config files, machine drivers should set"card.components" filed. component field with "cfg-spk:<> spk:<> cfg-hp:<hp> cfg-mic/cfg-dmics:". Based on this CRAS has to re-construct a string to load ucm files. If this component field is not set, then CRAS will have to load regular card name based ucm files.

I summarized based on my understanding.. @plbossart @cujomalainey : please see if this is line with expectation.

Reply:
@sathya-nujella thanks for the explanation. With unibuild this should not be an issue since we can install per unit ucms by redirecting cras with suffixes to the ucms.

Reply:
Thank you @cujomalainey . Based on this, I will drop patch:3 and we do not need to keep snd card name overriding patch.